### PR TITLE
Update dialyxir

### PR DESCRIPTION
**Pull request by MattMSumner**

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,3 @@
 rust 1.26.0
-elixir 1.7.2
-erlang 20.0
+elixir 1.7.2-otp-21
```
Comment (MattMSumner) [inline .tool-versions]: I've updated these to match what we have on circle to try and reproduce locally running dialyzer without emitting any warnings.

File: mix.exs [hunk 2]
```diff
@@ -18,6 +18,8 @@ defmodule Mana.MixProject do
       start_permanent: Mix.env() == :prod,
       dialyzer: [
         ignore_warnings: ".dialyzer.ignore-warnings",
+        plt_add_apps: [:mix],
+        plt_add_deps: :transitive,
```
Comment (MattMSumner) [inline mix.exs]: This could be removed. My understanding is that it's ensuring we include types from dependencies.

File: mix.exs [hunk 3]
```diff
@@ -18,6 +18,8 @@ defmodule Mana.MixProject do
       start_permanent: Mix.env() == :prod,
       dialyzer: [
         ignore_warnings: ".dialyzer.ignore-warnings",
+        plt_add_apps: [:mix],
```
Comment (MattMSumner) [inline mix.exs]: This was added to remove some annoying warnings about mix tasks.

File: .tool-versions (hunk 1 above)
Comment (ayrat555) [reply .tool-versions]: I think it's wrong because we can now only use otp 20

Comment (ayrat555) [reply .tool-versions]: til https://github.com/urbint/rox/pull/15 is merged

Comment (germsvel) [reply .tool-versions]: The weird thing is that CI uses otp-21 and all the tests pass. I guess the tests are using `ets` instead of `rox`?

Comment (ayrat555) [reply .tool-versions]: yep @germsvel

**Pull request merged**